recipe-client-addon/test/unit/test_PreferenceExperiments.js

```diff
@@ -477,3 +478,45 @@ add_task(withMockExperiments(withMockPreferences(function* testInit(experiments,
     "init set the value for a default pref experiment",
   );
 })));
+
+// init should register telemetry experiments
+add_task(withMockExperiments(async function testInit(experiments) {
+  experiments["experiment1"] = experimentFactory({name: "experiment1", branch: "branch1"});
+  experiments["experiment2"] = experimentFactory({name: "experiment2", branch: "branch2"});
+
+  let currentExperiments = TelemetryEnvironment.getActiveExperiments();
+  ok(!("experiment1" in currentExperiments), "experiment1 is not in telemetry");
+  ok(!("experiment2" in currentExperiments), "experiment2 is not in telemetry");
+
+  await PreferenceExperiments.init();
+
+  currentExperiments = TelemetryEnvironment.getActiveExperiments();
+  deepEqual(currentExperiments["experiment1"], {branch: "branch1"}, "experiment1 is in telemetry");
+  deepEqual(currentExperiments["experiment2"], {branch: "branch2"}, "experiment2 is in telemetry");
+}));
```

Comment:
Are the started experiments persisted across tests? We might want to clean up after ourselves so that other test cases aren't affected.